### DISCUSSION TITLE
Run dashboard from outside cluster?

### DISCUSSION THREAD

**Discussion by FGasper** [kubernetes/dashboard#7512] [General]
Hello,

Is it possible to run Kubernetes Dashboard from a machine outside the cluster but that has the right kube.config file to administer the cluster? If so, is there any documentation about doing this?

Thank you!

**Comment by maciaszczykm** [comment]
Hello @FGasper!

Simplest option would be to run Docker image with args matching your setup. You can find list of supported args [here](https://github.com/kubernetes/dashboard/blob/master/docs/common/dashboard-arguments.md) (including `kubeconfig`).

You can also check how we do it now after we have moved API and UI to separate containers [here](https://github.com/kubernetes/dashboard/blob/master/Makefile#L61). Note that it's master version, not 2.7.0 anymore.

Let us know if you have any questions.

**Reply by floreks** [reply]
I'd not recommend trying to run our new images (API and UI) since they need some gateway to "stitch" them together. It's best to use the latest official release (single container). Technically we support the `kubeconfig` flag that allows you to run it outside of the cluster but it is not an officially supported way of running Dashboard. We mostly use this for development and testing.

**Reply by maciaszczykm** [reply]
Yes, totally agree. I've linked it just to show how Docker compose/run command may look like.

**Comment by FGasper** [comment]
Thank you for your responses, @floreks and @maciaszczykm!

Our setup is thus: we ship a legacy (non-containerized) application that runs on a dedicated server. Users have UNIX system accounts, sometimes with shell access. This application shares network storage (Ceph or NFS, depending on the install) with a k8s cluster that runs the users’ websites. The legacy application administers the cluster via kubectl.

We’d like to run Dashboard as part of that legacy application. There can be multiple concurrent administrator sessions, and they would both need access to the Dashboard. Is `kubectl proxy` the best approach here? It seemed to me a bit simpler if we could deploy Dashboard from the legacy server, using the existing kubeconfig as authentication mechanism, but “I don’t know what I don’t know”.

**Comment by FGasper** [comment]
Additional question: if I want to proxy the Dashboard from, say, /some/custom/docroot instead of /, is that possible?

I don’t see any configuration to this effect in the Helm chart: https://github.com/kubernetes/dashboard/blob/master/charts/helm-chart/kubernetes-dashboard/values.yaml

**Comment by FGasper** [comment]
Answer to self: yes, it’s possible via `kubectl proxy`’s `--www-prefix` parameter.
